### PR TITLE
Handle skipped blobs side car for empty blobs in blobsSidecarsByRange

### PR DESCRIPTION
**Pull request by g11tech**
Handle skipped blobs side car for empty blobs in blobsSidecarsByRange
see: 
- https://github.com/ethereum/consensus-specs/pull/3174

### CODE REVIEW

File: packages/beacon-node/src/network/network.ts [hunk 1]
```diff
@@ -243,22 +244,53 @@ export class Network implements INetwork {
...
+          if (blobKzgCommitmentsLen !== 0) {
+            throw Error(
+              `Missing blobsSidecar for blockSlot=${block.message.slot} with blobKzgCommitmentsLen=${blobKzgCommitmentsLen}`
+            );
+          }
+          blobsSidecar = {
+            beaconBlockRoot: this.config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message),
+            beaconBlockSlot: block.message.slot,
+            blobs: [],
+            kzgAggregatedProof: ckzg.computeAggregateKzgProof([]),
```
Comment (dapplion) [inline packages/beacon-node/src/network/network.ts]: Why do you need to create an empty BlobsSidecar?

Comment (g11tech) [reply packages/beacon-node/src/network/network.ts]: because empty BlobsSidecar may be skipped in the req/resp, thats whats happening in devnet3

**Review by dapplion — COMMENTED** [review]
Given that this function is growing in complexity would make sense to move it to a file with signature

```ts
function beaconBlocksMaybeBlobsByRange(config, reqResp, peerId, request) {}
```

And unit test it by injecting a mock reqResp that resolves with predefined data

**Review by twoeths — APPROVED** [review]

File: packages/beacon-node/test/unit/network/doBeaconBlocksMaybeBlobsByRange.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import sinon, {SinonStubbedInstance} from "sinon";
+import {expect} from "chai";
+import {peerIdFromString} from "@libp2p/peer-id";
+import {ssz, eip4844} from "@lodestar/types";
+import {createIBeaconConfig, createIChainForkConfig, defaultChainConfig} from "@lodestar/config";
+
+import {doBeaconBlocksMaybeBlobsByRange, ReqRespBeaconNode} from "../../../src/network/reqresp/index.js";
+import {BlockInputType} from "../../../src/chain/blocks/types.js";
+
+describe("doBeaconBlocksMaybeBlobsByRange", function () {
```
Comment (dapplion) [inline packages/beacon-node/test/unit/network/doBeaconBlocksMaybeBlobsByRange.ts:10]: Please always append `.test.ts` extension to test files

Comment (g11tech) [reply packages/beacon-node/test/unit/network/doBeaconBlocksMaybeBlobsByRange.ts:10]: ahh thats my bad, here is the fix: https://github.com/ChainSafe/lodestar/pull/4949

**Pull request merged**